### ISSUE TITLE
Bikeshed discussion: package name

### ISSUE THREAD

**Issue by markerikson** [reduxjs/redux-starter-kit#36]
Time for a lovely bikeshed discussion on what exactly we should name this package!

At the moment, the repo is located under my Github (`markerikson/redux-starter-kit`), and published as a scoped NPM package under my username (`@acemarke/redux-starter-kit`).

I've always said that I want to turn this into an official Redux-branded package.  That first means moving it to our new `reduxjs` Github org, which is a no-brainer.  (In fact, could probably do that any time now.)

However, there's two questions beyond that:

- What should the actual name of the package itself be?
- How should we publish it on NPM - scoped or global package name?

There's already NPM packages called [`redux-starter-kit`](https://yarnpkg.com/en/package/redux-starter-kit) (an app boilerplate project) and [`redux-boost`](https://yarnpkg.com/en/package/redux-boost) (which is kinda similar to this package, but also does some data fetching stuff), so it looks like both of those are non-options.  (Unless we were to convince the owners to give us the package names or something.)

So, I'm open to suggestions for final package naming and publishing.

/cc @nickmccurdy , @sw-yx , @matthew-gerstman, and whoever else feels like commenting.

**Comment by nickmccurdy** [comment]
Moving to the reduxjs GitHub organization is clearly a win, though I think we should stick with unscoped packages to be consistent with Redux's own package name. Alternatively if you want to use npm organizations, we could move Redux to `@redux/core` like Babel did and then we can pick pretty much any name starting with `@redux/`. I would prefer a shorter package name, but if you prefer `redux-starter-kit` we could ask the package owner.

@timdorr I notice you own [redux-start](https://www.npmjs.com/package/redux-start) on npm and it appears to be parked. Could we use it or do you need it for something else?

**Comment by timdorr** [comment]
That's what I parked it for, so it's available to use.

I also have the @redux-js org on npm. Someone has @redux and @reduxjs orgs, so I grabbed what I could. If we want to get serious about it, I can reach to npm to see about taking over @redux (AFAICT, it's not in use, just squatted on).

**Comment by nickmccurdy** [comment]
Awesome, `redux-start` is my favorite package name so far. I think it would be best to either attempt to acquire the @redux org or continue using the current package names, as npm does not recommend including `js` or `node` in package names and org scopes are included in package names.

**Comment by JeffBaumgardt** [comment]
I would second `redux-start` or `redux-starter` and drop kit. I think npm has a policy in place for taking squatted names to official orgs. https://docs.npmjs.com/misc/disputes is once resource I found for this.

**Comment by markerikson** [comment]
Let's see if we can get `@redux` on NPM, on general principle, even if we don't immediately use it here.

**Comment by nickmccurdy** [comment]
I'm not against using the `@redux` org if we get it, but if we do we need to rename `redux` to `@redux/core` to prevent confusion.

**Comment by matthew-gerstman** [comment]
I vote we optimize for shipping. `redux-start` sounds like a perfectly reasonable name to me.

It doesn't seem like anyone here has a terribly strong opinion one way or the other and I'd rather start promoting an easy way to get started with redux than worry about what we call it.

**Comment by nickmccurdy** [comment]
@markerikson What do you think about switching to `redux-start` at least for now?

**Comment by hnordt** [comment]
I would like to suggest ReduxKit.

Inspiration: UIKit, ARKit, WatchKit, etc.

**Comment by nickmccurdy** [comment]
I like that name, but `redux-kit` is taken on npm.

**Comment by timdorr** [comment]
@markerikson Do you want to move this repo into @reduxjs and change the name to redux-start? If you can't do that for some reason, let me know and I'll fiddle with the knobs so you can.

**Comment by hnordt** [comment]
@nickmccurdy it's not `redux-kit`, it's `reduxkit`. 😄

**Comment by dandigangi** [comment]
`redux-starter-kit` sounds perfect and simple to me.

**Comment by markerikson** [comment]
Yep, the big problem with that is that the name is already taken on NPM: https://www.npmjs.com/package/redux-starter-kit .  We'd have to ask the author for it.

**Comment by timdorr** [comment]
`@redux-js/starter-kit`?

**Comment by timdorr** [comment]
@shotaK Can you donate the redux-starter-kit package name to us?

**Comment by markerikson** [comment]
I just sent a dispute email to NPM asking about possibly getting `@redux` or `@reduxjs`.  Got an auto-response saying the request had been received.

**Comment by nickmccurdy** [comment]
afaik npm generally waits 2 weeks, and if the owner doesn't respond, they give you the namespace

**Comment by shotaK** [comment]
@timdorr Done.

**Comment by markerikson** [comment]
Awesome.  @shotaK, thank you _very_ much - really appreciate it!

Okay, in that case, I'm inclined to just go with `redux-starter-kit` as the name, because A) the name is now available, and B) it's the smallest actual change involved (just gotta change the publishing details).

**Comment by markerikson** [comment]
Oooo-kay, I just hit the button to transfer this repo to the `reduxjs` org, and it looks like the transfer has completed.

I suppose we can debate a bit further whether we would want `@redux/starter-kit` or `redux-starter-kit`, if anyone has any strong opinions on that topic.

**Comment by dandigangi** [comment]
I feel like this is silly to say because I don't have a specific reason beyond familiarity bias, but I do think scoped packages should be left to addons/plugins/exts/etc. I think this is worthy of it's own distinct repo.

**Comment by markerikson** [comment]
@timdorr : awright, works for me!

In that case, I'll declare the issue as settled unless a compelling reason comes up to change it.

### Long live `redux-starter-kit` !

:)

**Comment by nickmccurdy** [comment]
We can transition later, but from what I've seen from Babel 7, there's quite a lot of overhead, especially if your package is frequently used as a peerDependency.

Are there plans to use the `@redux` org for `redux` (would be `@redux/core`) or other core packages?

**Comment by timdorr** [comment]
Not at the moment. I don't want to plan out anything that's contingent on the opaque decision process within npm, Inc.

**Comment by nickmccurdy** [comment]
Sorry I think I got confused, do we still not have `@redux`?

**Comment by markerikson** [comment]
No, we don't.  Tim said he has `@redux-js`.  `@redux` and `@reduxjs` are currently squatted, and this afternoon I sent in a dispute email asking about getting them.

**Comment by nickmccurdy** [comment]
Package/org names with `js` aren't recommended anyway, so I'm fine with `redux-starter-kit`. I'm glad we got it, thanks @shotaK!
